app.js

```diff
@@ -58,6 +58,24 @@ app.use((err, req, res, _next) => {
   res.status(err.status || 500).send(errorMessage);
 });
 
-app.listen(config.httpPort, () => {
+const server = app.listen(config.httpPort, () => {
   console.log(`HTTP server listening on port ${config.httpPort}.`);
 });
+
+
+function gracefulExit() {
+  console.log('Received exit request. Closing app...');
+
+  new Promise((resolve, reject) => {
+    server.close((err) => (err ? reject(err) : resolve()));
+  }).then(() => notificationsSender.stop())
+    .then(() => database.close())
+    .then(() => process.exit())
+    .catch((err) => {
```

Comment:
@julienw  One problem I thought of today. If one of the first steps fails, we don't execute the following ones and just exit the process. I don't know how much trouble this could make, but I wanted to call it out, explicitly.